google/cloud/compute/CMakeLists.txt

```diff
@@ -116,114 +175,154 @@ set(service_dirs
+    "region_operations/v1/" "zone_operations/v1/")
 
 include(GoogleCloudCppCommon)
-
 include(CompileProtos)
 google_cloud_cpp_find_proto_include_dir(PROTO_INCLUDE_DIR)
-
 google_cloud_cpp_load_protodeps(
-    proto_deps
-    "${PROJECT_SOURCE_DIR}/external/googleapis/protodeps/compute.deps")
+    deps "${PROJECT_SOURCE_DIR}/external/googleapis/protodeps/compute.deps")
 
 file(
-    GLOB proto_files
+    GLOB internal_proto_files
     LIST_DIRECTORIES false
-    RELATIVE_PATH "${CMAKE_CURRENT_SOURCE_DIR}" ${proto_globs})
-list(SORT proto_files)
+    RELATIVE_PATH "${CMAKE_CURRENT_SOURCE_DIR}" "v1/internal/*.proto")
+list(SORT internal_proto_files)
 
-google_cloud_cpp_proto_library(
-    google_cloud_cpp_compute_protos ${proto_files} PROTO_PATH_DIRECTORIES
-    "${EXTERNAL_GOOGLEAPIS_SOURCE}" "${PROTO_INCLUDE_DIR}"
-    "${PROJECT_SOURCE_DIR}")
-external_googleapis_set_version_and_alias(compute_protos)
-target_link_libraries(google_cloud_cpp_compute_protos PUBLIC ${proto_deps})
-# Some files are too big for the MSVC defaults.
-if (MSVC)
-    target_compile_options(google_cloud_cpp_compute_protos PRIVATE "/bigobj")
-endif ()
+compute_proto_library("google_cloud_cpp_compute_internal_protos" PROTO_FILES
+                      "${internal_proto_files}" DEPS "${deps}")
 
-file(
-    GLOB source_files
-    RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
-    ${source_globs})
-list(SORT source_files)
-add_library(google_cloud_cpp_compute ${source_files})
-target_include_directories(
-    google_cloud_cpp_compute
-    PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
-           $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
-           $<INSTALL_INTERFACE:include>)
-target_link_libraries(
-    google_cloud_cpp_compute
-    PUBLIC google-cloud-cpp::grpc_utils
-           google-cloud-cpp::common
-           google-cloud-cpp::rest_internal
-           google-cloud-cpp::rest_protobuf_internal
-           google-cloud-cpp::compute_protos)
-google_cloud_cpp_add_common_options(google_cloud_cpp_compute)
-set_target_properties(
-    google_cloud_cpp_compute
-    PROPERTIES EXPORT_NAME google-cloud-cpp::experimental-compute
-               VERSION "${PROJECT_VERSION}"
-               SOVERSION "${PROJECT_VERSION_MAJOR}")
-target_compile_options(google_cloud_cpp_compute
-                       PUBLIC ${GOOGLE_CLOUD_CPP_EXCEPTIONS_FLAG})
+list(APPEND deps "google_cloud_cpp_compute_internal_protos")
+unset(compute_proto_lib_targets)
+foreach (dir IN LISTS service_dirs operation_service_dirs)
+    if (NOT "${dir}" STREQUAL "v1/")
+        string(REPLACE "/v1/" "" short_dir "${dir}")
+        file(
+            GLOB ${short_dir}_proto_files
+            LIST_DIRECTORIES false
+            RELATIVE_PATH "${CMAKE_CURRENT_SOURCE_DIR}" "${dir}*.proto")
+        list(SORT ${short_dir}_proto_files)
+        compute_proto_library(
+            "google_cloud_cpp_compute_${short_dir}_protos" PROTO_FILES
+            "${${short_dir}_proto_files}" DEPS ${deps})
```

Comment:
... we could just say `DEPS "google_cloud_cpp_compute_internal_protos"` here and they all get picked up transitively.

This also applies below in a few places.